python/metatensor-operations/metatensor/operations/manipulate_dimension.py

```diff
@@ -418,30 +418,46 @@ def rename_dimension(tensor: TensorMap, axis: str, old: str, new: str) -> Tensor
     for block in tensor.blocks():
         samples = block.samples
         properties = block.properties
+        components = block.components
 
         if axis == "samples":
             samples = samples.rename(old, new)
         elif axis == "properties":
             properties = properties.rename(old, new)
+        elif axis == "components":
+            new_components: List[Labels] = []
+            for component in components:
+                if old in component.names:
+                    component = component.rename(old, new)
+                new_components.append(component)
+            components = new_components
```

Comment:
```suggestion
            components: List[Labels] = []
            for component in components:
                if old in component.names:
                    component = component.rename(old, new)
                components.append(component)
```

Reply:
Left as is to avoid overwriting components with an empty list